pkg/actors/actors.go

```diff
@@ -499,9 +499,6 @@ func (a *actorsRuntime) drainRebalancedActors() {
 					}
 				}
 
-				// don't allow state changes
-				a.actorsTable.Delete(key)
```

Comment:
@1046102779 Thanks for your PR. I looked into the history for this line and found this PR: https://github.com/dapr/dapr/pull/390

I am concerned about merging this PR for the following reasons:
1. There is a comment right above this line that makes me thing this is needed - it might seem duplicate but the fact that it is done here and not later in the processing can make a difference in concurrency calls.
2. If this does not work for some rare race condition, we will not know right away. Our E2E tests might not expose this.
3. The fact that changing this line did not break any unit test is also concerning.


Usually we don't need to be that skeptical about changes in Dapr but this part is very sensitive. Since it does not solve any issue, I don't think is worth the risk.